### PR TITLE
Fix type hint of `optuna/integration/sklearn.py`

### PR DESCRIPTION
**Pull request by akihironitta**
## Motivation
See #711. 

## Description of the changes
Fixed type hint of `optuna/integration/sklearn.py`. (#1062 tried to fix before.)

### CODE REVIEW

**Comment by akihironitta** [comment]
Does anyone know how to fix import order error from `flake8`? https://circleci.com/gh/optuna/optuna/54617

**Comment by zchenry** [comment]
I think the block of imports where `flake8` fails should be reordered as follows.

```python
from optuna._experimental import experimental
from optuna._imports import try_import
from optuna import distributions
from optuna import logging
from optuna import samplers
from optuna import study as study_module
from optuna.study import StudyDirection
from optuna.trial import FrozenTrial
from optuna.trial import Trial
from optuna import TrialPruned
```

Note that the actually imported modules should follow the alphabet order.

**Comment by akihironitta** [comment]
@zchenry Thank you so much for your help!

File: optuna/integration/sklearn.py
Comment (ytsmiling) [inline optuna/integration/sklearn.py:17]: ``pandas`` is not listed in the requirements of ``optuna``. Please remove this line and add the following block before the import block of ``sklearn``.

```python

with try_import():
    import pandas as pd
```

File: optuna/integration/sklearn.py
Comment (ytsmiling) [inline optuna/integration/sklearn.py:56]: After moving ``pandas`` to ``try_import`` block, ``pandas`` might not available here. So please use ``"pd.Seriese"`` and ``"pd.DataFrame"``instead of ``pd.Seriese`` and ``pd.DataFrame``, respectively.

File: optuna/integration/sklearn.py
Comment (ytsmiling) [inline optuna/integration/sklearn.py:660]: Ditto.

**Review by ytsmiling — COMMENTED** [review]
Thank you for your contribution. I left some comments to resolve the failing CI.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR @akihironitta and thanks for the advice @ytsmiling! LGTM except for @ytsmiling's comments.

File: optuna/integration/sklearn.py
Comment (akihironitta) [reply optuna/integration/sklearn.py]: Thank you for reviewing. Applied!

File: optuna/integration/sklearn.py
Comment (ytsmiling) [inline optuna/integration/sklearn.py:32]: When ``pandas`` is not installed, python exits this block before importing ``sklearn`` even if ``sklearn`` is installed. I recommend splitting the try_import block as follows.

```python
with try_import:
    import pandas as pd

with try_import() as _imports:
    ...
```

**Review by ytsmiling — CHANGES_REQUESTED** [review]

File: optuna/integration/sklearn.py
Comment (akihironitta) [reply optuna/integration/sklearn.py:32]: I see. Thank you again for detailed comments.

**Comment by ytsmiling** [comment]
Thank you for addressing the comment @akihironitta .
And sorry for the confusion @akihironitta , but after adding the review comment, I changed my mind, and I now think that
1. We should not split the import block, and
2. We should replace ``BaseValidator`` in type hints with ``"BaseValidator"``.
What do you think @HideakiImamura ?

**Comment by HideakiImamura** [comment]
Thanks for the kind reviews @ytsmiling.

> 1. We should not split the import block, and
> 2. We should replace BaseValidator in type hints with "BaseValidator".

IMO, we should not sprint the import block. Avoiding the CI error by using "BaseValidator" is reasonable to me.

@akihironitta Sorry for the confusion. Could you apply the above changes?

**Comment by akihironitta** [comment]
@HideakiImamura @ytsmiling 
> 2. We should replace BaseValidator in type hints with "BaseValidator".

Do you mean to replace `BaseEstimator` and `BaseCrossValidator` with `"BaseEstimator"` and `"BaseCrossValidator"`, respectively, in case `sklearn` (or `pandas`) is unavailable?

**Review by ytsmiling — APPROVED** [review]
Oops, nice catch concerning ``BaseEstimator``.
Thank you for your contribution, again. LGTM!

**Review by HideakiImamura — APPROVED** [review]
Thanks for your swift actions! LGTM.

**Comment by HideakiImamura** [comment]
@hvy Could you merge this PR after your approval?

**Review by hvy — APPROVED** [review]
LGTM! Sorry for holding out on the PR, thanks a bunch for the rather large change.

**Pull request merged**
